mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,6 +49,41 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
+		public static X509CertificateImpl InitFromHandle (IntPtr handle)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return InitFromHandleApple (handle);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+
+#if MONO_FEATURE_APPLETLS // If we support AppleTls, which is the default, and not overriding to legacy
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return InitFromHandleApple (handle);
+#endif
+#if !MOBILE
+			return InitFromHandleCore (handle);
+#elif !MONOTOUCH && !XAMMAC
+			throw new NotSupportedException ();
+#endif
+#endif
```

Comment:
The code path that didn't return a value meant we promoted that build configuration (that I don't think could happen right now) into a build error (from a runtime exception).